src/libcore/marker.rs

```diff
@@ -92,10 +111,37 @@ impl<T: ?Sized> !Send for *mut T { }
 #[lang = "sized"]
 #[rustc_on_unimplemented = "`{Self}` does not have a constant size known at compile-time"]
 #[fundamental] // for Default, for example, which requires that `[T]: !Default` be evaluatable
+#[cfg(not(stage0))]
+pub trait Sized: ?Move {
+    // Empty.
+}
+
+/// docs
+#[stable(feature = "rust1", since = "1.0.0")]
+#[lang = "sized"]
+#[rustc_on_unimplemented = "`{Self}` does not have a constant size known at compile-time"]
+#[fundamental] // for Default, for example, which requires that `[T]: !Default` be evaluatable
+#[cfg(stage0)]
 pub trait Sized {
     // Empty.
 }
 
+/// Types that can be moved after being borrowed.
+#[cfg(not(stage0))]
+#[lang = "move"]
+#[unstable(feature = "immovable_types", issue = "0")]
+pub unsafe trait Move: ?Move {
```

Comment:
weird: do you have to have `?Move` constraint on this trait?

Just curious, since I noticed that `trait Sized` does not have an analogous `?Sized` bound...

Reply:
Auto traits cannot have supertraits because that is unsound. Also we do not want `Move` to require `Move`. That would probably result in interesting things happening.